### ISSUE TITLE
[idea] Make the ex-mode input field a palette similar to command palette.

### ISSUE THREAD

**Issue by trusktr** [lloeki/ex-mode#150]
The way the command palette works is awesome.

Idea:

Make ex-mode similar to that, but also similar to [advanced-open-file](https://atom.io/packages/advanced-open-file)'s palette (it has awesome autocompletion of directory and file names).

But, as a bonus, this palette could be positioned at the bottom, and the command completions, etc, would go vertically upwards (instead of downwards like the default command palette).

So basically, the current behavior would remain exactly the same, but added would be an upward command palette with file/directory completion similar to advanced-open-file.

**Comment by LongLiveCHIEF** [comment]
The idea of `ex-mode` is to pair with `vim-mode`, to create the experience of editing with VIM in atom. One of the things about VIM that is great, is the lack of all the UI extra's that tend to complicate the experience of editing a file.

However, if you like this experience, `ex-mode` just maps commands (`:w`) to actions... you could create a custom keymap to accomplish this behavior very easily.

**Comment by trusktr** [comment]
> One of the things about VIM that is great, is the lack of all the UI extra's that tend to complicate the experience of editing a file.

That's kind of like arguing to just go back to Vim and not using Atom.  I believe there is a way to make this happen without complicating the experience text editing, and in fact making it better.

Current ex-mode doesn't have completion (at least not out-of-the-box, as pressing tab closes the ex-mode and tabs my text, which is arguably not a nice text editing experience). Having completion (without the added UI) would be better than current. Maybe that could be something to start with (for example, doing it like `advanced-open-file` does, but without the UI).
